src/v/s3/client.cc

```diff
@@ -515,7 +517,10 @@ client_pool::client_pool(
 
 ss::future<> client_pool::stop() {
     _as.request_abort();
-    return _gate.close();
+    _cvar.broken();
+    // Wait until all leased objects are returned
+    co_await _gate.close();
+    co_return co_await _cvar.wait([this] { return _pool.size() == _size; });
```

Comment:
this looks a little odd here. waiting on a broken condition variable would immediately throw, no?

Reply:
It looks like `broken` doesn't act like gate in this regard, subsequent calls to `wait` and `signal` are possible after `broken` was called.

Reply:
Fixed that by removing the need to wait here.